src/context.rs

```diff
@@ -399,26 +402,59 @@ impl<A: Actor> Context<A> {
         Ok(fut)
     }
 
-    fn inbox(&self) -> impl FusedStream<Item = Either<BroadcastMessage<A>, AddressMessage<A>>> {
-        let addr_rx = self.receiver.clone();
-        let broadcast_rx = self.broadcast_receiver.clone();
+    fn inbox(&self) -> Inbox<A> {
+        Inbox::new(self.receiver.clone(), self.broadcast_receiver.clone())
+    }
+}
+
+struct Inbox<A> {
+    inner: BoxStream<'static, Either<BroadcastMessage<A>, AddressMessage<A>>>,
```

Comment:
small nit: it might be less overhead and allocation to not use a box and just store what we need inline and then manually implement stream - can see how the benches look

Reply:
Would you mind if we optimize these things later?

Reply:
Yea, if the peformance isn't too bad then we can leave it for now